### DISCUSSION TITLE
Assembler: Make "PLC14500-Nano can only JMP 0" a warning instead of an error

### DISCUSSION THREAD

**Discussion by wrh3c** [nicolacimmino/PLC-14500#4] [Ideas]
In the assembler, I recommend making the "PLC14500-Nano can only JMP 0" a warning rather than an error.  This will allow the assembler to generate all 256 possible opcodes.  A JMP with a non-zero argument will still jump to 0 on the PLC14500-Nano, since nothing is done with the 4-bit argument, but the 4-bit argument can be seen on the DATA LEDs when using the slow clock mode or single-stepping.

**Comment by nicolacimmino** [comment]
"A JMP with a non-zero argument will still jump to 0", that's what worried me. Someone will write a program expecting it to do something but will then be left scratching their heads on why it just keeps resetting the PC.

**Reply by wrh3c** [reply]
Right, so an assembler warning would be appropriate.  The JMP argument is only 4 bits while the address bus is 8 bits, so it's not obvious how the argument would correlate to an address anyway.  My recommendation is so that all 256 possible opcodes can be generated - we could feed the output of a disassembler into the assembler and get the original byte code.  This is similar to my other suggestion about having LEDs for FLAGF, FLAGO, and RTN - the PLC14500-Nano doesn't do anything with these signals, but the assembler allows the NOPF, NOPO, and RTN instructions with 4-bit arguments.

**Reply by nicolacimmino** [reply]
You convinced me ;) 

https://github.com/nicolacimmino/PLC-14500/commit/c212e6fe934ed5fdb21eb2b5f21ffb30736acc7f
